src/util/CancellationHandle.h

```diff
@@ -33,15 +33,24 @@ enum class CancellationState {
 
 namespace detail {
 
-#if !defined(QLEVER_DISABLE_QUERY_CANCELLATION_WATCH_DO) && LOGLEVEL >= WARN
-constexpr bool ENABLE_QUERY_CANCELLATION_WATCH_DOG = true;
-#else
+/// Helper enum that selects which features to compile inside the
+/// `CancellationHandle` class.
+enum class CancellationMode { ENABLED, NO_WATCH_DOG, DISABLED };
+
+#if !defined(QUERY_CANCELLATION_MODE) || QUERY_CANCELLATION_MODE == ENABLED
+constexpr CancellationMode QUERY_CANCELLATION_MODE = CancellationMode::ENABLED;
+#elif QUERY_CANCELLATION_MODE == DISABLED
+constexpr CancellationMode QUERY_CANCELLATION_MODE = CancellationMode::DISABLED;
+#elif QUERY_CANCELLATION_MODE == NO_WATCH_DOG
+constexpr CancellationMode QUERY_CANCELLATION_MODE =
+    CancellationMode::NO_WATCH_DOG;
 #if LOGLEVEL < WARN
 #warning \
     "QLEVER_ENABLE_QUERY_CANCELLATION_WATCH_DOG is ignored if WARN logging \
 level is disabled"
 #endif
-constexpr bool ENABLE_QUERY_CANCELLATION_WATCH_DOG = false;
+#else
+#error "Invalid value for QUERY_CANCELLATION_MODE!"
```

Comment:
Okay, depends on whether you want to leave it with those checks, then you can leave it as is...